<a href="https://colab.research.google.com/github/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis/blob/main/Air_BNB_2019_V1_EDA_Submission_Template.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name**    -



##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Team Member 1 -**
##### **Team Member 2 -**
##### **Team Member 3 -**
##### **Team Member 4 -**

# **Project Summary -**

Write the summary here within 500-600 words.
Story of what you have done in the whole project not like conclusion of

# **GitHub Link -**

**Python-EDA-Air-BNB-2019-Analysis repository**

https://github.com/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis

# **Problem Statement**


**Air BNB Dataset Analysis**

Since 2008, guests and hosts have used Airbnb to expand on travelling possibilities and present a more unique, personalised way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analysed and used for security, business decisions, understanding of customers' and providers' (hosts) behaviour and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. This dataset has around 49,000 observations in it with 16 columns and it is a mix of categorical and numeric values. Explore and analyse the data to discover key understandings.

#### **Define Your Business Objective?**

Analysis and generating insights from the data, helping in understanding of customers' and providers' (hosts) behaviour and performance on the platform, guiding marketing initiatives, implementation of secure and innovative additional services and much more

Data oriented business decisions, helps in improving processes and services, translating the complex data and discover key understanding for future actionable plans.

# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [ ]:
# Import Libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


### Dataset Loading

In [ ]:
# Load Dataset
# Loading the repository files from the github to collab
!git clone https://github.com/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis.git

# Giving a path to the file
path = "/content/Python-EDA-Air-BNB-2019-Analysis/Airbnb NYC 2019.csv"

Cloning into 'Python-EDA-Air-BNB-2019-Analysis'...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 6 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (6/6), 2.25 MiB | 10.78 MiB/s, done.


In [ ]:
# Read/load the dataset from the loaded repository files
dataset = pd.read_csv(path)
dataset

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

### Dataset First View

In [ ]:
# Dataset First Look
dataset.head()

### Dataset Rows & Columns count

In [ ]:
# Dataset Rows & Columns count
dataset.shape

(48895, 16)

### Dataset Information

In [ ]:
# Dataset Info
# Getting info about the datatype column wise dtypes: float64(3 column), int64(7 column), object(6 column)
# & non null values present in the data
# last_review is in object datatype, it should be in datetime dtype
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

#### Duplicate Values

In [ ]:
# Drop the unique id column
dataset.drop(["id"], axis = 1, inplace = True)

In [ ]:
# Dataset Duplicate Value Count
print("Number of duplicate rows :",dataset.duplicated().sum())

dataset.drop_duplicates(inplace=True)

Number of duplicate rows : 0


In [ ]:
# Dataset Duplicate Value Count
# check host id
duplicates = dataset["host_id"].duplicated()
dataset[duplicates]

name    host_id  \
22                            CBG Helps Haiti Room#2.5      22486   
23                               CBG Helps Haiti Rm #2      22486   
33                                 back room/bunk beds      32294   
35     Lovely room 2 & garden; Best area, Legal rental       9744   
39                 LowerEastSide apt share shortterm 1       7549   
...                                                ...        ...   
48887  Garden Jewel Apartment in Williamsburg New York  208514239   
48889                               QUIT PRIVATE HOUSE  107716952   
48890  Charming one bedroom - newly renovated rowhouse    8232441   
48891    Affordable room in Bushwick/East Williamsburg    6570630   
48893             43rd St. Time Square-cozy single bed   30985759   

             host_name neighbourhood_group       neighbourhood  latitude  \
22               Lisel            Brooklyn          Park Slope  40.67989   
23               Lisel            Brooklyn          Park Slope  40.68001   
33     Ssameer Or Trip           Manhattan              Harlem  40.82130   
35              Laurie            Brooklyn         South Slope  40.66869   
39                 Ben           Manhattan     Lower East Side  40.71401   
...                ...                 ...                 ...       ...   
48887            Melki            Brooklyn        Williamsburg  40.71232   
48889          Michael              Queens             Jamaica  40.69137   
48890          Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891          Marisol            Brooklyn            Bushwick  40.70184   
48893              Taz           Manhattan      Hell's Kitchen  40.75751   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
22     -73.97798     Private room     80               1                 39   
23     -73.97865     Private room    110               2                 71   
33     -73.95318     Private room     50               3                273   
35     -73.98780     Private room     89               4                168   
39     -73.98917      Shared room     40               1                214   
...          ...              ...    ...             ...                ...   
48887  -73.94220  Entire home/apt    170               1                  0   
48889  -73.80844     Private room     65               1                  0   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48893  -73.99112      Shared room     55               1                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
22     2019-01-01               0.37                               6   
23     2019-07-02               0.61                               6   
33     2019-07-01               2.37                               3   
35     2019-06-21               1.41                               3   
39     2019-07-05               1.81                               4   
...           ...                ...                             ...   
48887         NaN                NaN                               3   
48889         NaN                NaN                               2   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48893         NaN                NaN                               6   

       availability_365  
22                  364  
23                  304  
33                  359  
35                  340  
39                  188  
...                 ...  
48887               365  
48889               163  
48890                 9  
48891                36  
48893                 2  

[11438 rows x 15 columns]

In [ ]:
# Dataset Duplicate Value Count
# check name of the listing
duplicates1 = dataset["name"].duplicated()
dataset[duplicates1]

name    host_id  \
330                    Loft w/ Terrace @ Box House Hotel     417504   
339                                 Superior @ Box House     417504   
580                   ✿✿✿ COUNTRY COTTAGE IN THE CITY✿✿✿    1138692   
661                                  The Lenox in Harlem    1313306   
669                     Loft Suite @ The Box House Hotel     417504   
...                                                  ...        ...   
48684                                       Comfort home  266211707   
48735  Sunny, Cozy, Private Room In The Heart of Bush...  147515897   
48759                                    Home Sweet Home  273656890   
48791                                Home away from home  238163900   
48826                                  Brooklyn's finest   66084717   

                 host_name neighbourhood_group    neighbourhood  latitude  \
330    The Box House Hotel            Brooklyn       Greenpoint  40.73842   
339    The Box House Hotel            Brooklyn       Greenpoint  40.73813   
580           Keera (Jena)           Manhattan  Lower East Side  40.71892   
661                 Yvette           Manhattan           Harlem  40.81122   
669    The Box House Hotel            Brooklyn       Greenpoint  40.73641   
...                    ...                 ...              ...       ...   
48684                  Yan            Brooklyn      Sunset Park  40.64439   
48735               Flávia            Brooklyn         Bushwick  40.70366   
48759                Liana           Manhattan      East Harlem  40.79266   
48791                 Lucy              Queens  Cambria Heights  40.68557   
48826                  Tim            Brooklyn    East Flatbush  40.65170   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
330    -73.95312     Private room    249               3                  2   
339    -73.95394     Private room    179               3                 13   
580    -73.98401  Entire home/apt    199               1                 14   
661    -73.94279  Entire home/apt    400               5                  0   
669    -73.95330  Entire home/apt    199               3                 24   
...          ...              ...    ...             ...                ...   
48684  -74.01816     Private room    185               1                  0   
48735  -73.92728     Private room     84               3                  0   
48759  -73.94740     Private room     50               1                  0   
48791  -73.72731     Private room     50               3                  0   
48826  -73.92580  Entire home/apt    200               3                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
330    2011-05-12               0.02                              28   
339    2019-06-27               0.14                              28   
580    2015-07-20               0.15                               2   
661           NaN                NaN                               2   
669    2018-11-06               0.32                              28   
...           ...                ...                             ...   
48684         NaN                NaN                               2   
48735         NaN                NaN                               1   
48759         NaN                NaN                               1   
48791         NaN                NaN                               1   
48826         NaN                NaN                               2   

       availability_365  
330                  60  
339                  81  
580                 158  
661                 365  
669                  84  
...                 ...  
48684               177  
48735                28  
48759                81  
48791               176  
48826               206  

[989 rows x 15 columns]

#### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
dataset.isnull().sum()

name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [ ]:
# Visualizing the missing values


### What did you know about your dataset?

Here we are having the datset of Air BNB 2019, we have to analyse the data and discover insights to provide better understanding of the requirements & facilitate decision making and give more appropriate & innovative services to our clients.

There are no duplicate rows in our dataset, duplicate host id values represents that individual host can post different rooms or apartment.

While duplicate name values represents that the listing could be a hotel or house having many different rooms availability.

The data is having 48895 rows, 16 columns. There are some null values in name, host_name,last_review and reviews_per_month column & non null values present in the data.

Last_review is in object datatype, while should be in datetime dtype

The last_review and reviews_per_month column may have null values because of "0" number_of_reviews at that destination .

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns
dataset.columns

Index(['name', 'host_id', 'host_name', 'neighbourhood_group', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'price', 'minimum_nights',
       'number_of_reviews', 'last_review', 'reviews_per_month',
       'calculated_host_listings_count', 'availability_365'],
      dtype='object')

In [ ]:
# Dataset Describe
dataset.describe(include = "all")

name       host_id host_name neighbourhood_group  \
count            48879  4.889500e+04     48874               48895   
unique           47905           NaN     11452                   5   
top     Hillside Hotel           NaN   Michael           Manhattan   
freq                18           NaN       417               21661   
mean               NaN  6.762001e+07       NaN                 NaN   
std                NaN  7.861097e+07       NaN                 NaN   
min                NaN  2.438000e+03       NaN                 NaN   
25%                NaN  7.822033e+06       NaN                 NaN   
50%                NaN  3.079382e+07       NaN                 NaN   
75%                NaN  1.074344e+08       NaN                 NaN   
max                NaN  2.743213e+08       NaN                 NaN   

       neighbourhood      latitude     longitude        room_type  \
count          48895  48895.000000  48895.000000            48895   
unique           221           NaN           NaN                3   
top     Williamsburg           NaN           NaN  Entire home/apt   
freq            3920           NaN           NaN            25409   
mean             NaN     40.728949    -73.952170              NaN   
std              NaN      0.054530      0.046157              NaN   
min              NaN     40.499790    -74.244420              NaN   
25%              NaN     40.690100    -73.983070              NaN   
50%              NaN     40.723070    -73.955680              NaN   
75%              NaN     40.763115    -73.936275              NaN   
max              NaN     40.913060    -73.712990              NaN   

               price  minimum_nights  number_of_reviews last_review  \
count   48895.000000    48895.000000       48895.000000       38843   
unique           NaN             NaN                NaN        1764   
top              NaN             NaN                NaN  2019-06-23   
freq             NaN             NaN                NaN        1413   
mean      152.720687        7.029962          23.274466         NaN   
std       240.154170       20.510550          44.550582         NaN   
min         0.000000        1.000000           0.000000         NaN   
25%        69.000000        1.000000           1.000000         NaN   
50%       106.000000        3.000000           5.000000         NaN   
75%       175.000000        5.000000          24.000000         NaN   
max     10000.000000     1250.000000         629.000000         NaN   

        reviews_per_month  calculated_host_listings_count  availability_365  
count        38843.000000                    48895.000000      48895.000000  
unique                NaN                             NaN               NaN  
top                   NaN                             NaN               NaN  
freq                  NaN                             NaN               NaN  
mean             1.373221                        7.143982        112.781327  
std              1.680442                       32.952519        131.622289  
min              0.010000                        1.000000          0.000000  
25%              0.190000                        1.000000          0.000000  
50%              0.720000                        1.000000         45.000000  
75%              2.020000                        2.000000        227.000000  
max             58.500000                      327.000000        365.000000

### Variables Description

Answer Here

In [ ]:
# availability_365 have 0 values that does not make sense as there should be availability to book rooms
# have to check are hosts not putting their rooms on airbnb

### Check Unique Values for each variable.

In [ ]:
# Check Unique Values for each variable.

unique_counts = dataset.nunique()
print(unique_counts)

name                              47905
host_id                           37457
host_name                         11452
neighbourhood_group                   5
neighbourhood                       221
latitude                          19048
longitude                         14718
room_type                             3
price                               674
minimum_nights                      109
number_of_reviews                   394
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
dtype: int64


## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.

# Drop duplicate rows
dataset.drop_duplicates(inplace=True)

# all the rows having Nan values in name column
dataset[dataset["name"].isnull()== True]

# Removing Null values in name column
dataset.dropna(subset=['name'], inplace =True)


# Removing unnecessary columns
dataset.drop(["host_name","latitude","longitude"], axis = 1, inplace =True)


# Replacing null values that are in last_review and reviews_per_month column
dataset.fillna(0, inplace = True)


dataset.isnull().sum()


KeyError: ignored

In [ ]:
# change last review datatype

### What all manipulations have you done and insights you found?

Answer Here.

## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1

In [ ]:
# Chart - 1 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 2

In [ ]:
# Chart - 2 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 3

In [ ]:
# Chart - 3 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 4

In [ ]:
# Chart - 4 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 5

In [ ]:
# Chart - 5 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 6

In [ ]:
# Chart - 6 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 7

In [ ]:
# Chart - 7 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 8

In [ ]:
# Chart - 8 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 9

In [ ]:
# Chart - 9 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 10

In [ ]:
# Chart - 10 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 11

In [ ]:
# Chart - 11 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12

In [ ]:
# Chart - 12 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 13

In [ ]:
# Chart - 13 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 14 - Correlation Heatmap

In [ ]:
# Correlation Heatmap visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

#### Chart - 15 - Pair Plot

In [ ]:
# Pair Plot visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***